menas/src/main/scala/za/co/absa/enceladus/menas/services/VersionedModelService.scala

```diff
@@ -72,6 +72,66 @@ abstract class VersionedModelService[C <: VersionedModel with Product with Audit
     versionedMongoRepository.getLatestVersionValue(name)
   }
 
+  def exportSingleItem(name: String, version: Int): Future[String] = {
+    getVersion(name, version).flatMap({
+      case Some(item) => Future(item.exportItem())
+      case _ => throw NotFoundException()
+    })
+  }
+
+  def exportLatestItem(name: String): Future[String] = {
+    getLatestVersion(name).flatMap({
+      case Some(item) => Future(item.exportItem())
+      case _ => throw NotFoundException()
+    })
+  }
+
+  def importSingleItem(item: C, username: String, metadata: Map[String, String]): Future[Option[C]] = {
+    for {
+      validation <- validateSingleImport(item, metadata)
+      result <- {
+        if (validation.isValid()) {
+          importItem(item, username)
+        } else {
+          throw ValidationException(validation)
+        }
+      }
+    } yield result
+  }
+
+  private[services] def validateSingleImport(item: C, metadata: Map[String, String]): Future[Validation] = {
```

Comment:
Why is the return type `Future[]`? IMHO just unnecessary complicates things all around.

Reply:
When you look at the overrides of this method, you can see it cannot return non-Future validations.

Reply:
I still don't see. Can we have a call on that?

Reply:
I was wrong on getting rid of the `Future`, sorry. Of course `Await` makes no sense here.